torch/distributed/checkpoint/examples/stateful_example.py

```diff
@@ -0,0 +1,104 @@
+# Owner(s): ["oncall: distributed"]
```

Comment:
We can later move this to https://github.com/pytorch/examples or update this tutorial in https://github.com/pytorch/tutorials/blob/main/recipes_source/distributed_checkpoint_recipe.rst. 

From what I learned, we should not keep the examples PT repo, but we still have a couples for checkpoint and other distributed componenets.